tests/framework/src/test.rs

```diff
@@ -15,6 +15,26 @@ pub struct Font {
     pub italic: bool,
 }
 
+#[derive(Clone, Copy)]
+pub enum TestKind {
+    Slow,
```

Comment:
What I meant before is, why is this "kind" type even needed?
Do you have any other "kinds" of tests you plan to add in the near future, that are mutually exclusive with them being "slow"? If not, I feel like this is just premature over engineering - YAGNI.
Or is this somehow a technical requirement to make test ordering with this attribute possible/easier?

Reply:
Oh sure, using test kinds was actually the easiest way. It's not like "I created test kinds to possibly support more than one kind", but rather "libtest_mimic supports test kinds and I can utilize that for slow tests and test ordering". I think I could create a wrapper for a test trial that contains the information about the test ordering, but wouldn't _that_ be over engineering?

Compare that approach also to other frameworks, for instance:
* JUnit 4 test categories: https://github.com/junit-team/junit4/wiki/Categories (the example mentions `SlowTests.class` category),
* JUnit 5 tags: https://junit.org/junit5/docs/current/user-guide/#writing-tests-tagging-and-filtering (the example mentions a `fast` tag),
* pytest markers: https://docs.pytest.org/en/7.1.x/example/markers.html (docs mention a `slow` test mark).

The big disadvantage of libtest_mimic test kinds is that a test cannot have more than one kind, but it is possible to store multiple kinds by joining them e.g. with a comma (`kind1,kind2`).

Using test kinds also means that tests with a kind are shown as such in the output (i.e. slow tests are prefixed with `[slow]`).

If you're referring to the enum and not test kinds in general, well... that's only a high-level representation of a kind. It's better IMO than comparing strings, and it also encapsulates ordering logic. If you have any suggestions on improving it, I'm all ears!

Reply:
> but rather "libtest_mimic supports test kinds and I can utilize that for slow tests and test ordering".

Ooh, right, pardon my ignorance... It's been a while since we discussed this... ^^'